controllers/add_pods.go

```diff
@@ -117,7 +118,7 @@ func (a addPods) reconcile(ctx context.Context, r *FoundationDBClusterReconciler
 			pod.Annotations[fdbv1beta2.PublicIPAnnotation] = ip
 		}
 
-		err = r.PodLifecycleManager.CreatePod(ctx, r, pod)
+		err = r.PodLifecycleManager.CreatePod(logr.NewContext(ctx, logger), r, pod)
```

Comment:
I think it makes sense to create the new context with the logger in the cluster controller e.g. here https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/cluster_controller.go#L96 and just assign that new context to the `ctx` variable that get's passed down into all `reconcile` methods. That way we are able to reduce the `log.WithValues` calls to only add the reconciler information.

This should be adde in another PR (I can create an issue later to track those efforts). Just wanted to. discuss that idea.